desafio-02/guiribeirodev/javascript/desafio-02.js

```diff
@@ -0,0 +1,23 @@
+// Função "verificaPrimo()"" irá verificar se o número é primo, a verificação é feita com base na premissa que todo número primo irá ter apenas dois divisores, o número 1 e ele próprio, então caso o número de divisores sejam 2 será primo.
+
+//Função "listaPrimo" é responsável por criar a lista dos números primos do número inicial até o numero final.
+
+function listaPrimo(numeroInicio, numeroFim) {
+  for (numeroInicio = 1; numeroInicio <= numeroFim; numeroInicio++) {
+    verificaPrimo(numeroInicio)
+  }
+}
+
+function verificaPrimo(numeroInicio) {
+  let divisor = 0
+
+  for (let count = 1; count <= numeroInicio; count++) {
```

Comment:
Tem uma condição adicional que pode colocar neste for além de `count <= numeroInicio` que pode aumentar bastante a performance do teu programa.

Fiz o teste no console do Chrome do teu programa e ele está batendo 565 milisegundos, segue o código que usei para o benchmark:
```js
let t0 = performance.now()
listaPrimo(1, 10000)
let t1 = performance.now()

console.log('Time taken: ' + (t1 - t0) + ' millis')
```

Quando adicionei a condição, o tempo caiu mais da metade, para 160-180 milisegundos. Não vou te dar a resposta, mas a dica está na linha 18 do teu código.

Reply:
Dei uma ajustada na disposição dos comentários, transformei o comentário grande em um comentário de multi-linhas e dei umas quebras de linha. Quanto a performance eu adicionei duas novas condições o que reduziu bastante o tempo, testando no node pelo terminou ficou entre 95 -105ms, bem melhor que originalmente, obrigado pelo feedback!